utils/ixdtf/src/parsers/records.rs

```diff
@@ -56,28 +58,24 @@ pub struct TimeRecord {
     pub minute: u8,
     /// A second value.
     pub second: u8,
-    /// A millisecond value
-    pub millisecond: u16,
-    /// A Microsecond value
-    pub microsecond: u16,
-    /// A Nanosecond value
-    pub nanosecond: u16,
+    /// A nanosecond value representing all sub-second components.
+    pub nanosecond: u32,
 }
 
 /// `TimeZone` data
 #[non_exhaustive]
-#[derive(Default, Debug, Clone, PartialEq)]
-pub struct TimeZone<'a> {
+#[derive(Debug, Clone, PartialEq)]
+pub enum TimeZoneRecord<'a> {
     /// TimeZoneIANAName
-    pub name: Option<&'a str>,
+    Name(&'a str),
     /// TimeZoneOffset
-    pub offset: Option<UTCOffset>,
+    Offset(UTCOffsetRecord),
 }
 
 /// A full precision `UtcOffset`
 #[non_exhaustive]
 #[derive(Debug, Clone, Copy, PartialEq)]
-pub struct UTCOffset {
+pub struct UTCOffsetRecord {
     /// The `+`/`-` sign of this `UtcOffset`
```

Comment:
Nit: do one of the following

1. Document what the output values are. If the number is positive, is `sign > 0`? `sign >= 0`? `sign == 1`?
2. Switch to using an enum
3. Switch to using a bool

Reply:
Assuming you also use this in DurationParseRecord, I think it's probably worth making it an enum.